crates/ruff_cli/tests/integration_test.rs

```diff
@@ -300,6 +300,178 @@ fn nursery_direct() {
     -:1:2: E225 Missing whitespace around operator
     Found 1 error.
 
+    ----- stderr -----
+    warning: Selection of nursery rule `E225` without the `--preview` flag is deprecated.
+    "###);
+}
+
+#[test]
+fn nursery_group_selector() {
+    // Only nursery rules should be detected e.g. E225 and a warning should be displayed
+    let args = ["--select", "NURSERY"];
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(STDIN_BASE_OPTIONS)
+        .args(args)
+        .pass_stdin("I=42\n"), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    -:1:1: CPY001 Missing copyright notice at top of file
+    -:1:2: E225 Missing whitespace around operator
+    Found 2 errors.
+
+    ----- stderr -----
+    warning: The `NURSERY` selector has been deprecated. Use the `--preview` flag instead.
+    "###);
+}
+
+#[test]
+fn nursery_group_selector_preview_enabled() {
+    // Only nursery rules should be detected e.g. E225 and a warning should be displayed
+    let args = ["--select", "NURSERY", "--preview"];
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(STDIN_BASE_OPTIONS)
+        .args(args)
+        .pass_stdin("I=42\n"), @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    -:1:1: CPY001 Missing copyright notice at top of file
+    -:1:2: E225 Missing whitespace around operator
+    Found 2 errors.
+
+    ----- stderr -----
+    warning: The `NURSERY` selector has been deprecated. Use the `PREVIEW` selector instead.
+    "###);
+}
+
+#[test]
+fn preview_enabled_prefix() {
+    // E741 and E225 (preview) should both be detected
```

Comment:
What will happen to this test when E225 gets promoted to stable? (i don't have good answer for testing this)

Reply:
Great question, do we choose a new code? Should we have a couple of rules in a "TEST" category that do nothing?

Reply:
I'm looking into this

Reply:
Not working... but an idea https://github.com/astral-sh/ruff/compare/zanie/rule-warning...zanie/rule-tests

Reply:
I'm a bit hesitant about adding test-only rules. We would need to filter them out in documentation, schema generation and prevent they can be selected in production (but then, we want them to be selectable in tests). 

We could add the category behind a rust feature and make this an integration test (integration tests can require features), but this still suffers from the above-mentioned problems, although it has some more safeguards around it. 

To me this is an inherent downside of our statically generated rule schema (instead of deriving a `Rule` array that has somewhat weaker keys which would allow you to mock out the rules).

I don't have a good solution to propose that doesn't require changing our rule structure or abstracting some of the logic by a trait that then allows overriding whether a rule is preview or not.

Reply:
Doesn't the `#[cfg(test)]` approach address most of those concerns?

Reply:
We can also wait and see how much of a burden it is to maintain the tests as-is. I don't like it though.

Reply:
iirc `#[cfg(test)]` is only available in the same crate and not across crates (https://users.rust-lang.org/t/what-are-the-rules-for-cfg-test/54122/2), so e.g. `ruff` `cfg(test)` rules would not be available in `ruff_cli`.

Reply:
Resources:
- https://nrxus.github.io/faux/guide/exporting-mocks.html
- https://github.com/rust-lang/cargo/issues/8379